packages/langium/src/utils/promise-utils.ts

```diff
@@ -0,0 +1,34 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+export type MaybePromise<T> = T | Promise<T>
+
+/**
+ * Delays the execution of the current code to the next tick of the event loop.
```

Comment:
Add a comment that this shouldn't be used in a loop to avoid creating too many promises?